src/entry.rs

```diff
@@ -525,17 +551,29 @@ impl<'a> EntryFields<'a> {
 
         // Ensure we write a new file rather than overwriting in-place which
         // is attackable; if an existing file is found unlink it.
-        fn open(dst: &Path) -> io::Result<std::fs::File> {
-            OpenOptions::new().write(true).create_new(true).open(dst)
+        let mode = self.header.mode();
+        fn open(dst: &Path, _mode: &io::Result<u32>) -> io::Result<std::fs::File> {
+            let mut options = OpenOptions::new();
+            #[cfg(windows)]
+            {
+                // On windows create the file with the desired attributes rather
+                // than setting them in a separate syscall.
+                if readonly(_mode) {
+                    // File is requested to be readonly
+                    // 0x1 is FILE_ATTRIBUTE_READONLY in the Win32 API
```

Comment:
Why not make it a const instead of a comment here?

Reply:
Thats from #202